corehq/apps/api/resources/v0_5.py

```diff
@@ -312,7 +319,7 @@ def obj_create(self, bundle, **kwargs):
                 email=bundle.data.get('email', '').lower(),
             )
             del bundle.data['password']
-            self._update(bundle)
+            self._update(bundle, raise_error_if_invalid_key=False)
```

Comment:
Is `bundle.data` just a dict? We could copy it and then pop off the domain, username, password and email keys before passing it to `_update`.

That would then raise an exception if there were additional keys in the data which I think is better than silently ignoring them.

Reply:
I didn't copy the dict since it isn't used after this call and that is how the `password` field is handled currently. I also didn't include the email and domain fields because the _update method has logic to account for when a value hasn't changed, and I don't think there is a domain field in this dictionary.